app/assets/javascripts/components/overview/milestones.jsx

```diff
@@ -28,19 +31,26 @@ const Milestones = createReactClass({
     const weekNumberOffset = CourseDateUtils.weeksBeforeTimeline(this.props.course);
     const blocks = [];
 
-    this.props.allWeeks.map((week) => {
+    
+    this.props.allWeeks.map((week,navIndex) => {
       if (week.empty) return null;
-
       const milestoneBlocks = filter(week.blocks, block => block.kind === this.milestoneBlockType);
       return milestoneBlocks.map((block) => {
+        if (
```

Comment:
It looks we're doing something similar in both this component and the Timeline, checking for whether to map a particular block to `null`. Is this something that can become shared logic in the selector, or do they handle these cases differently?

Reply:
Actually, the problem I encountered is that in the selector, I needed the index for the weeks to display, so I used `weeks` in the `const getWeeks = state => state.timeline.weeks;` function to memoize all the results. However, the issue is that the `allWeekDates` being sent from `getAllWeekDates(state)` contains weeks that are after the timeline end date, but the `getWeeks` function doesn't include them. 

To handle this, when creating the `DateCalculator` instance:

```javascript
const dateCalc = new DateCalculator(course.timeline_start, course.timeline_end, index, {
  zeroIndexed: true
});
```

I found that the last indexes didn't have any dates, causing it to be empty when trying to render in the milestones and timeline. To fix this, I added a check for that in both files.